sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRequest.java

```diff
@@ -88,6 +116,21 @@ public HttpRequest setUrl(URL url) {
         return this;
     }
 
+    /**
+     * Set the target address to send the request to.
+     *
+     * @param url target address as a String
+     * @return this HttpRequest
+     */
+    public HttpRequest setUrl(String url) {
+        try {
+            this.url = new URL(url);
+        } catch (MalformedURLException e) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'url' must be a valid URL."));
```

Comment:
Same here. Add `e` as an argument to the `IllegalArgumentException` constructor.